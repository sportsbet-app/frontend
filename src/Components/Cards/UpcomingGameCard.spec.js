import React from 'react'
import { render } from '@testing-library/react'
//import renderer for snapshot
import renderer from 'react-test-renderer'
//import img for test
import nflColorImg2 from '../../Assets/nflColor2.jpg'
import nflColorImg from '../../Assets/nflColor.jpg'

import UGameCard from './UpcomingGameCard'

//create test data for the rendering
const data = {
  first: [{ strTeamBadge: { nflColorImg2 } },
  { strTeam: 'title1' }],
  second: [{ strTeamBadge: { nflColorImg } },
  { strTeam: 'title2' }],
  date: Date.now(),
  time: '3pm eastern'
}

//test to check that the app will display when called.
describe('<UGameCard />', () => {
  it('should display', () => {
    let load = render(<UGameCard />)

  })

  //create a snapshot
  it('matches snapshot', () => {
    const tree = renderer.create(<UGameCard gameInfo={data} />); // generates a DOM tree

    // snapshots are a JSON representation of the DOM tree
    expect(tree.toJSON()).toMatchSnapshot();
  });
})
