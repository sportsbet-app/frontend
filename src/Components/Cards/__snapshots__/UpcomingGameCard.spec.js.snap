// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UGameCard /> matches snapshot 1`] = `
<div
  className="sc-bdVaJa anjUL"
>
  <div
    className="sc-bwzfXH fKkCEQ"
  >
    <img
      alt="undefined"
      className="sc-bxivhb grZmpA"
      title="undefined"
    />
    <div>
      <h3>
        VERSUS
      </h3>
      <h3>
        1567267574871
         | 
        3pm eastern
      </h3>
    </div>
    <img
      alt="undefined"
      className="sc-bxivhb grZmpA"
      title="undefined"
    />
  </div>
  <div
    className="sc-htpNat hitxPo"
  >
    <div
      className="cardContentRow"
    >
      <button
        className="sc-ifAKCX bVBizH"
      >
        +211
      </button>
      <button
        className="sc-ifAKCX bVBizH"
      >
        +211
      </button>
      <button
        className="sc-ifAKCX bVBizH"
      >
        +211
      </button>
    </div>
    <div
      className="cardContentRow"
    >
      <button
        className="sc-ifAKCX bVBizH"
      >
        +211
      </button>
      <button
        className="sc-ifAKCX bVBizH"
      >
        +211
      </button>
      <button
        className="sc-ifAKCX bVBizH"
      >
        +211
      </button>
    </div>
  </div>
</div>
`;
